daemon/daemon_linux.go

```diff
@@ -237,3 +239,11 @@ func supportsRecursivelyReadOnly(cfg *configStore, runtime string) error {
 	}
 	return fmt.Errorf("rro is not supported by runtime %q", runtime)
 }
+
+// apiSocket returns a Unix socket for container to access the daemon's API
+func (daemon *Daemon) apiSocket(_ string, opts *dmount.APISocketOptions) (string, error) {
+	if opts != nil && opts.Access != dmount.AccessUnconfined {
+		return "", fmt.Errorf("API socket access %q is not supported", opts.Access)
+	}
```

Comment:
This will not error out if we don't pass any options at all:


```
curl -X POST -H "Content-Type: application/json"   --unix-socket /var/run/docker.sock   http://localhost/containers/create   -d '{
    "Image": "alpine",
    "Cmd": ["sleep", "3600"],
    "HostConfig": {
       "Mounts": [
        {
          "Type": "apisocket"
        }
      ]
    }
  }'
```